torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -5370,10 +5370,17 @@ std::tuple<Tensor, Tensor> scatter_reduce_backward(
     grad_self = grad;
     grad_src = grad.gather(dim, index);
   } else if (reduce == "prod") {
-    grad_self = (grad * result) / self;
-    grad_self.masked_fill_(self == 0, 0);
-    grad_src = (grad * result).gather(dim, index) / src;
-    grad_src.masked_fill_(src == 0, 0);
+    // Explicitly compute exclusive prod for elements in self/src that are 0
```

Comment:
Could we re-use the logic from `prod_backward` (in this same file) here?

Reply:
Hm, I'm not quite sure how we would use cumprod here to calculate the exclusive product given that there's a scatter operation.

Reply:
It might be harder to re-use that in the scatter case indeed because you don't have a `cumprod` version of scatter reduce?
It was just a high level question to know if we can de-dupe code there. Also the cumprod version makes it easy to ignore the problems with 0s :p